go/gildedrose/gildedrose_test.go

```diff
@@ -1,19 +1,72 @@
-package gildedrose_test
+package gildedrose
 
-import (
-	"testing"
+import "testing"
 
-	"github.com/emilybache/gildedrose-refactoring-kata/gildedrose"
-)
-
-func Test_Foo(t *testing.T) {
-	var items = []*gildedrose.Item{
-		{"foo", 0, 0},
+func TestUpdateQuality(t *testing.T) {
+	type args struct {
+		items []*Item
 	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "handle empty items",
+			args: args{items: []*Item{}},
+		},
+		{
+			name: "handle Brie",
+			args: args{
+				items: []*Item{
+					{Quality: 1, Name: "Brie", SellIn: 1}, {Quality: 2, Name: "Brie", SellIn: 0},
+				},
+			},
+		},
+		{
+			name: "handle Tafkal",
+			args: args{
+				items: []*Item{
+					{Quality: 1, Name: "Tafkal", SellIn: 1}, {Quality: 2, Name: "Tafkal", SellIn: 0},
+				},
+			},
+		},
+		{
+			name: "handle Sulf",
+			args: args{
+				items: []*Item{
+					{Quality: 1, Name: "Sulf", SellIn: 1}, {Quality: 2, Name: "Sulf", SellIn: 0},
+				},
+			},
+		},
+		{
+			name: "handle Other",
+			args: args{
+				items: []*Item{
+					{Quality: 1, Name: "Other", SellIn: 1}, {Quality: 2, Name: "Other", SellIn: 0},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			UpdateQuality(tt.args.items)
```

Comment:
What are you testing here? There are no assertions...

Before writing tests here, please watch https://www.youtube.com/watch?v=tWn8RA_DEic and make sure the tests are named VERY well (in a way that's like a Spec - if I read them I 100% understand the requirements).